docs/cookbook/entities/custom-model-accessible-per-admin-channel.rst

```diff
@@ -0,0 +1,392 @@
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    final class ChannelPerAdminChannelType extends AbstractType
+    {
+        /** @var RepositoryInterface */
+        private $channelRepository;
+
+        /** @var AdminChannelProviderInterface */
+        private $adminChannelProvider;
+
+        public function __construct(
+            RepositoryInterface $channelRepository,
+            AdminChannelProviderInterface $adminChannelProvider
+        ) {
+            $this->channelRepository = $channelRepository;
+            $this->adminChannelProvider = $adminChannelProvider;
+        }
+
+        public function buildForm(FormBuilderInterface $builder, array $options): void
+        {
+            if ($options['multiple']) {
+                $builder->addModelTransformer(new CollectionToArrayTransformer());
+            }
+        }
+
+        public function configureOptions(OptionsResolver $resolver): void
+        {
+            $resolver->setDefaults([
+                'choices' => function (Options $options): array {
+                    /** @var ChannelInterface|null $adminChannel */
+                    $adminChannel = $this->adminChannelProvider->getChannel();
+
+                    if ($adminChannel != null) {
+                        return [$adminChannel];
+                    }
+
+                    return $this->channelRepository->findAll();
+                },
+                'choice_value' => 'code',
+                'choice_label' => 'name',
+                'choice_translation_domain' => false,
+            ]);
+        }
+
+        public function getParent(): string
+        {
+            return ChoiceType::class;
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'sylius_channel_choice';
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Form\ChannelPerAdminChannelType:
+        arguments: ['@sylius.repository.channel', '@Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface']
+        tags: ['form.type']
+
+The ``Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface`` service allows getting a list of proper channels for the currently logged-in admin.
+
+* Next, create a form type for your entity:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Form;
+
+    use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+    use Symfony\Component\Form\Extension\Core\Type\TextType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    final class SupplierType extends AbstractResourceType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('name', TextType::class, [
+                    'label' => 'Name'
+                ])
+                ->add('channel', ChannelPerAdminChannelType::class, [
+                    'label' => 'Channel'
+                ])
```

Comment:
Couldn't we just add here:
```suggestion
                ->add('channels', ChannelChoiceType::class, [
                    'choices' => $this->availableChannelsForAdminProvider->getChannels(),
                    'label' => 'sylius.ui.channel',
                ])
```
instead of adding the entire `ChannelPerAdminChannelType` form type?